Python/gc_free_threading.c

```diff
@@ -1006,6 +1023,7 @@ gc_collect_internal(PyInterpreterState *interp, struct collection_state *state)
     _PyEval_StopTheWorld(interp);
     // merge refcounts for all queued objects
     merge_all_queued_objects(interp, state);
+    process_delayed_frees(interp);
```

Comment:
Nit/question: `_PyEval_StopTheWorld` also locks `_PyRuntime`. Maybe is worth moving the lock up?

Reply:
`_PyEval_StopTheWorld` locks `_PyRuntime` temporarily to loop over threads, but does not hold the lock

We want `process_delayed_frees` after the stop-the-world call because it guarantees that we can free all the "delayed free" pointers. If we moved it up before the `_PyEval_StopTheWorld` there might be some memory blocks we don't free.

Reply:
Makes sense, thanks for clarifying

Reply:
> If we moved it up before the `_PyEval_StopTheWorld` there might be some memory blocks we don't free.

With "moving the lock up" i meant to have a unique locking for both functions, not to move this call before the other. But your comment still applies